src/main/java/net/dv8tion/jda/internal/requests/restaction/PermissionOverrideActionImpl.java

```diff
@@ -97,19 +63,13 @@ public GuildChannel getChannel()
     @Override
     public Role getRole()
     {
-        return role;
+        return isRole() ? (Role) permissionHolder : null;
```

Comment:
Thoughts on throwing instead of returning null?

Reply:
I think its better to return null, this makes it easier to check whether an override is for a specific entity.

Example: 
`guild.getPublicRole().equals(override.getRole())`

vs

`override.isRole() && guild.getPublicRole().equals(override.getRole())`

Reply:
I do feel like the latter adds clarity, and throwing an error would result in people writing code that is less prone to having silent errors.

I don't feel strongly enough to forcibly require the change though.